txtdirect.go

```diff
@@ -181,51 +187,23 @@ func Redirect(w http.ResponseWriter, r *http.Request, c Config) error {
 
 	if isIP(host) {
 		log.Println("[txtdirect]: Trying to access 127.0.0.1, fallback triggered.")
-		fallback(w, r, "", "", "", 0, c)
+		fallback(w, r, "", "", "global", 0, c)
 		return nil
 	}
 
 	rec, err := getRecord(host, r.Context(), c, r)
 	if err != nil {
-		if strings.HasSuffix(err.Error(), "no such host") {
-			if c.Redirect != "" {
-				log.Printf("[txtdirect]: %s > %s", r.Host+r.URL.Path, c.Redirect)
-				w.Header().Add("Cache-Control", fmt.Sprintf("max-age=%d", status301CacheAge))
-				w.Header().Add("Status-Code", strconv.Itoa(http.StatusMovedPermanently))
-				http.Redirect(w, r, c.Redirect, http.StatusMovedPermanently)
-				if c.Prometheus.Enable {
-					RequestsByStatus.WithLabelValues(host, strconv.Itoa(http.StatusMovedPermanently)).Add(1)
-				}
-				return nil
-			}
-			if contains(c.Enable, "www") {
-				s := strings.Join([]string{defaultProtocol, "://", defaultSub, ".", host}, "")
-				log.Printf("[txtdirect]: %s > %s", r.Host+r.URL.Path, s)
-				w.Header().Add("Cache-Control", fmt.Sprintf("max-age=%d", status301CacheAge))
-				w.Header().Add("Status-Code", strconv.Itoa(http.StatusMovedPermanently))
-				http.Redirect(w, r, s, http.StatusMovedPermanently)
-				if c.Prometheus.Enable {
-					RequestsByStatus.WithLabelValues(host, strconv.Itoa(http.StatusMovedPermanently)).Add(1)
-				}
-				return nil
-			}
-			http.NotFound(w, r)
-			if c.Prometheus.Enable {
-				RequestsByStatus.WithLabelValues(host, strconv.Itoa(http.StatusNotFound)).Add(1)
-			}
+		if err != nil {
+			fallback(w, r, "", "", "global", http.StatusMovedPermanently, c)
```

Comment:
Let's use temporary for the global fallback as this might be due to a misconfiguration so temporary seems best here.